### DISCUSSION TITLE
How to import a large library

### DISCUSSION THREAD

**Discussion by patricksan** [TortugaPower/BookPlayer#881] [General]
I'm importing a book with 925 mp3.
Normally around 4,42 Gb.

I select import from my iCloud, but it doesn't work. Ik keeps freeze and I need to kill the app.
I tried import from 60 files each time and it works, but it will take a long time.

Is there another way to import a long book as that, please?

thank you

**Comment by GianniCarlo** [comment]
Hi @patricksan!, hmmm if AirDrop is not an option, I think the next best thing is having a `.zip` file with your books, and import that instead, the app will handle unzipping the files and importing them into the app. The Files app integration is sometimes unreliable when the books are not already downloaded into the device, as it fails sometimes to properly show the download progress, and it's difficult to know if it stopped or if it's really downloading things. 

For the next upcoming major version (v5.0.0) we've implemented a share extension, so you can go to the Files app directly, and share the files from there to BookPlayer (right now if you try it that way, it will only import one file sadly)

**Reply by patricksan** [reply]
Hi @GianniCarlo, I tried with AirDrop, but just one file was imported.

The option with a `.zip` was perfect. I added the huge zip in the iCloud and just imported into BookPlayer. Worked like a charm!


I appreciate your help and looking forward the major version coming.


Thanks for the great work with this app.
